### PR TITLE
Fix BiasDropoutFusion when there are multiple Dropout consumers.

### PR DESCRIPTION
**Pull request by kit1980**
If the Dropout has multiple consumers of output 0, the residual Add shouldn't be fused.

### CODE REVIEW

File: onnxruntime/core/optimizer/bias_dropout_fusion.cc [hunk 1]
```diff
@@ -15,49 +15,56 @@ void FuseResidualAddIfAny(Graph& graph, const Node& dropout_node,
...
-        dropout_input.push_back(residual_add_node.MutableInputDefs()[1]);  // residual
-      } else if (dropout_output_name == residual_add_node.InputDefs()[1]->Name()) {
-        dropout_input.push_back(residual_add_node.MutableInputDefs()[0]);  // residual
+  int dropout_consumers_count = 0;
+  for (auto edge_itr = dropout_node.OutputEdgesBegin(); edge_itr != dropout_node.OutputEdgesEnd(); ++edge_itr) {
+    if (edge_itr->GetSrcArgIndex() == 0) {
+      ++dropout_consumers_count;
+    }
+  }
+  // To be able to fuse the resudial Add,
```
Comment (mrry) [inline onnxruntime/core/optimizer/bias_dropout_fusion.cc:25]:
```suggestion
  // To be able to fuse the residual Add,
```

**Review by mrry — DISMISSED** [review]
Just one small typo, but this looks good to me!

File: onnxruntime/core/optimizer/bias_dropout_fusion.cc [hunk 2]
```diff
@@ -15,49 +15,56 @@ void FuseResidualAddIfAny(Graph& graph, const Node& dropout_node,
...
-      }
 
-      Node& residual_add_node = *graph.GetNode(last_node.Index());
-      const std::string& dropout_output_name = dropout_node.OutputDefs()[0]->Name();
-      if (dropout_output_name == residual_add_node.InputDefs()[0]->Name()) {
-        dropout_input.push_back(residual_add_node.MutableInputDefs()[1]);  // residual
-      } else if (dropout_output_name == residual_add_node.InputDefs()[1]->Name()) {
-        dropout_input.push_back(residual_add_node.MutableInputDefs()[0]);  // residual
+  int dropout_consumers_count = 0;
+  for (auto edge_itr = dropout_node.OutputEdgesBegin(); edge_itr != dropout_node.OutputEdgesEnd(); ++edge_itr) {
```
Comment (ashbhandare) [inline onnxruntime/core/optimizer/bias_dropout_fusion.cc:20]: you can use dropout_node.GetOutputEdgesCount() instead

**Review by codemzs — APPROVED** [review]
Thanks, @kit1980 ! Looks great!

File: onnxruntime/core/optimizer/bias_dropout_fusion.cc (hunk 2 above)
Comment (kit1980) [reply onnxruntime/core/optimizer/bias_dropout_fusion.cc:20]: `GetOutputEdgesCount()` will count all edges, even for the mask output, right?
I only need edges for the output 0.

Comment (ashbhandare) [reply onnxruntime/core/optimizer/bias_dropout_fusion.cc:20]: got it

**Pull request merged**